src/Domain/Structure/ElementId.cpp

```diff
@@ -142,6 +143,15 @@ bool operator<(const ElementId<VolumeDim>& lhs,
   return false;
 }
 
+template <size_t Dim>
+bool is_zeroth_element(const ElementId<Dim>& id) {
```

Comment:
It would helpful to take a `std::optional<size_t>` for the grid index, I think. Then if the optional is not set you don't compare the grid index (mostly what we will want to do in evolutions), while in the elliptic solver there are generally going to be multiple grids for the multigrid solver and so having the option to specify it would be helpful, I think